test/scripts/parallel-local/test-runner.js

```diff
@@ -0,0 +1,101 @@
+const { exec } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+let isFirstTest = true;
+
+/**
+ * Ensures the directory exists before executing tests.
+ * @param {string} testType The type of the test.
+ * @returns {void} Does not return a value.
+ */
+const ensureDirectoryExists = (testType) => {
```

Comment:
It does more than ensuring, it creates it so I would name it `createTestDirectoryIfNotExist`

Reply:
Done